Dockerfile

```diff
@@ -1,38 +1,53 @@
-FROM node:20.10-alpine
-
-MAINTAINER MoJ Digital, Probation in Court <probation-in-court-team@digital.justice.gov.uk>
-ARG BUILD_NUMBER
-ARG GIT_REF
+## use slim not alpine otherwise PACT binarys wont load due to missing libs
+FROM node:20.10-slim as base
 
+## common
 ENV TZ=Europe/London
 RUN ln -snf "/usr/share/zoneinfo/$TZ" /etc/localtime && echo "$TZ" > /etc/timezone
+RUN apt-get update
+RUN apt-get -y install g++ make python3
+WORKDIR /usr/src/app
+COPY package.json ./
+COPY package-lock.json ./
+RUN npm i
+EXPOSE 3000
 
+## only used for production
+FROM base as production
+MAINTAINER MoJ Digital, Probation in Court <probation-in-court-team@digital.justice.gov.uk>
+ARG BUILD_NUMBER
+ARG GIT_REF
 ENV BUILD_NUMBER ${BUILD_NUMBER:-1_0_0}
 ENV GIT_REF ${GIT_REF:-dummy}
-
-RUN apk add --no-cache g++ make python3 bash
-RUN mkdir -p /app
-WORKDIR /app
-
-COPY package*.json package-lock*.json ./
-
-RUN npm i
-
-COPY . .
-
+ENV APP_VERSION=${BUILD_NUMBER}
+ENV NODE_ENV='production'
+COPY ./server ./server
+COPY ./public ./public
+COPY ./bin ./bin
 RUN export APP_VERSION=${BUILD_NUMBER} && \
     export BUILD_NUMBER=${BUILD_NUMBER} && \
     export GIT_REF=${GIT_REF} && \
-    npm run css-build && \
-    npm run minify-js-css && \
-    npm run record-build-info
-
+    ./bin/build-css && \
+    ./bin/build-js && \
+    npx webpack --config ./public/config/webpack.config.js && \
+    ./bin/record-build-info
 RUN rm -rf node_modules && \
     npm ci --only=production --ignore-scripts
-
-EXPOSE 3000
-ENV APP_VERSION=${BUILD_NUMBER}
-ENV NODE_ENV='production'
 USER 2000
-
-CMD [ "node", "./bin/www" ]
+ENTRYPOINT [ "node" ]
+CMD [ "./bin/www" ]
+
+## only used for CI (in development they are mounted)
+FROM base as ci
+COPY ./server ./server
+COPY ./tests ./tests
+COPY ./schemas ./schemas
+## next statement used by PACT not wiremock!
+COPY ./wiremock ./wiremock
+ENTRYPOINT ["node"]
+
+## only use for development and integration tests
+FROM base as development
+RUN npm i -g concurrently
+ENTRYPOINT ["concurrently"]
+CMD [ "\"./node_modules/.bin/webpack --config ./public/config/webpack.config.js --watch\"", "\"./node_modules/.bin/nodemon --watch ./server ./bin/www\"", "\"./node_modules/.bin/nodemon --ext scss --watch ./public/src/stylesheets ./bin/build-css\"", "\"./node_modules/.bin/nodemon --ext js --watch ./public/src/javascripts ./bin/build-js\""]
```

Comment:
tbh you can move all this compiling of js/scss into webpack itself and just get webpack to hot reload etc but sure that will bloat this PR further

Reply:
The feature js files are separates, I didn't want to start refactoring application code as part of this work.